### DISCUSSION TITLE
Return weights in extract_spectrum_features

### DISCUSSION THREAD

**Discussion by wolski** [fgcz/depiction#4] [Q&A]
I suggest that we extend the return value of the `extract_spectrum_features` method of the `CalibrationMethod` interface as follows:

    def extract_spectrum_features(self, peak_mz_arr: NDArray[float], peak_int_arr: NDArray[float]) -> Tuple[DataArray, DataArray]:
        """Extracts a vector of features (dimension ["c"]) from a given, peak picked spectrum.
        For calibration methods which do not involve a feature extraction, an empty DataArray should be returned.
        :param peak_mz_arr: m/z values of the peaks in the spectrum
        :param peak_int_arr: intensity values of the peaks in the spectrum
        :return: a DataArray with the extracted features, with dimensions ["c"], DataArray with feature weights, with the same dimension as a feature, which can be used by the spatial smoothing algorithm.
        """

        feature_array = DataArray([], dims=["c"])
        weight_array = DataArray([], dims=["c"])
        return feature_array, weight_array

**Comment by leoschwarz** [comment]
It is already possible to put them into the feature array with the current xarray-based API, i.e. assuming we have 3 features and one weight.

`DataArray([f1, f2, f3, w], dims=["c"], coords={"c":["f1", "f2", "f3", "w"]})`

the benefit would be that it would be generic for most of the existing code. Implementing weighting into the smoothing step would anyhow yet have to be done.

then features can be accessed by `DataArray.sel`:

- weights: `array.sel(c="w")`
- feats: `array.sel(c=["f1", "f2", "f3"])` or `array.drop_sel(c="w")`
